### DISCUSSION TITLE
How to translate builder into other languages

### DISCUSSION THREAD

**Discussion by dpfr2022** [Budibase/budibase#9255] [Help] [answered]
Hello!

As far as I know (correct me if I'm wrong), there is currently no support for languages other than English in the builder.

According to this [link](https://github.com/Budibase/budibase/discussions/1150) localization of the builder is currently not a priority of the Budibase team.

Imagine I wanted to translate the builder into a different language (i. e. modify the source code of Budibase to support multiple languages).

Are there any recommendations (apart from ideas in [this discussion](https://github.com/Budibase/budibase/discussions/2023)) how this can be done so that it complies with the philosophy behind Budibase?

Thanks in advance

**Comment by melohagan** [comment, marked as answer]
Hey @dpfr2022 

One possibility is to have a lookup table with translated words used in your app for different languages. 
Possibly something like:

| Language (options picker) | Dictionary (JSON field) |
| --- | --- |
| Spanish | { "Name": "Nombre", etc } |
| French | { "Name": "Nom", etc } |

Then in your app screens you could have the following setup:

---

> Data provider (filtered on the Language, e.g. Spanish)
> > Repeater (to expose the singleton record returned from the filter)
> > > Form
> > > > Text input for Name field (label is bound to `{{ Repeater.Dictionary.Name }}`)

---

Another possibility is to use [custom components](https://docs.budibase.com/docs/custom-component). For example, I built a Date Picker that included support for preset locale: https://github.com/melohagan/budibase-component-kasper-date-picker

**Comment by dpfr2022** [comment]
@melohagan Thank you!
